src/project/Room.java

```diff
@@ -136,10 +136,9 @@ public Booking findAvailableDatesForIntervalAndSize(Booking interval) {
 
 
         for (int i = 0; i < bookedDates.size() - 1; i++) {
-            if (interval.getFrom().compareTo(bookedDates.get(i).getTo()) >= 0) {
-                return (new Booking(0L, bookedDates.get(i).getTo(), bookedDates.get(i + 1).getFrom(), null));
-            } else
+            if (interval.getFrom().compareTo(bookedDates.get(i).getTo()) > 0) {
```

Comment:
compareTo method is from Comparable, it's more readable to use the isAfter isBefore